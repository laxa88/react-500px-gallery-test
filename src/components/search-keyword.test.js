import React from 'react';
import 'jest';
import 'enzyme';

import SearchKeyword from './search-keyword';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('SearchKeyword', () => {
  test('can render', () => {
    const wrapper = shallow(
      <SearchKeyword
        onBlur={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('can trigger change and callback', () => {
    const callback = jest.fn();

    const wrapper = shallow(
      <SearchKeyword
        onBlur={callback}
      />
    );

    const expectedValue = 'dummyValue';

    wrapper.instance().handleChange({
      target: {value: expectedValue},
    });

    wrapper.instance().handleBlur();

    expect(callback).toBeCalledWith(expectedValue);
  });
});
