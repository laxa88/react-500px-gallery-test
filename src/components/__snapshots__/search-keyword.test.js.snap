// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchKeyword can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchKeywordInput
    onBlur={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Filter by keyword ",
        <input
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      "Filter by keyword ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onBlur": [Function],
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Filter by keyword ",
          <input
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          />,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        "Filter by keyword ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onBlur": [Function],
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
